lib/pbkit/pbkit.c

```diff
@@ -2287,183 +2287,126 @@ void pb_end(DWORD *pEnd)
-#endif
-
-    *(p+0)=EncodeMethod(subchannel,command,3);
+    pb_pushto(subchannel,p,command,3);
     *(p+1)=param1;
     *(p+2)=param2;
     *(p+3)=param3;
 }
 
-void pb_push4to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3, DWORD param4)
+inline DWORD *pb_push4to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3, DWORD param4)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push4to : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push4to : missing pb_begin earlier\n");
-    pb_PushIndex+=5;
-    pb_PushNext+=5;
-    if (pb_PushIndex>128) debugPrint("pb_push4to: begin-end block musn't exceed 128 dwords\n");
-#endif
-
-    *(p+0)=EncodeMethod(subchannel,command,4);
+    pb_pushto(subchannel,p,command,4);
     *(p+1)=param1;
     *(p+2)=param2;
     *(p+3)=param3;
     *(p+4)=param4;
 }
 
-
-void pb_push1(DWORD *p, DWORD command, DWORD param1)
+inline void pb_push(DWORD *p, DWORD command, DWORD nparam)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push1: new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push1: missing pb_begin earlier\n");
-    pb_PushIndex+=2;
-    pb_PushNext+=2;
-    if (pb_PushIndex>128) debugPrint("pb_push1: begin-end block musn't exceed 128 dwords\n");
-#endif
+    pb_pushto(SUBCH_3D,p,command,nparam);
+}
 
-    *(p+0)=EncodeMethod(SUBCH_3D,command,1);
+inline DWORD *pb_push1(DWORD *p, DWORD command, DWORD param1)
+{
+    pb_push(p,command,1);
     *(p+1)=param1;
+    return p+2;
 }
 
-void pb_push2(DWORD *p, DWORD command, DWORD param1, DWORD param2)
+inline DWORD *pb_push2(DWORD *p, DWORD command, DWORD param1, DWORD param2)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push2 : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push2 : missing pb_begin earlier\n");
-    pb_PushIndex+=3;
-    pb_PushNext+=3;
-    if (pb_PushIndex>128) debugPrint("pb_push2: begin-end block musn't exceed 128 dwords\n");
-#endif
-
-    *(p+0)=EncodeMethod(SUBCH_3D,command,2);
+    pb_push(p,command,2);
     *(p+1)=param1;
     *(p+2)=param2;
+    return p+3;
 }
 
-void pb_push3(DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3)
+inline DWORD *pb_push3(DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push3 : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push3 : missing pb_begin earlier\n");
-    pb_PushIndex+=4;
-    pb_PushNext+=4;
-    if (pb_PushIndex>128) debugPrint("pb_push3: begin-end block musn't exceed 128 dwords\n");
-#endif
-
-    *(p+0)=EncodeMethod(SUBCH_3D,command,3);
+    pb_push(p,command,3);
     *(p+1)=param1;
     *(p+2)=param2;
     *(p+3)=param3;
+    return p+4;
```

Comment:
This stuff can all be reduced to `return pb_push3to(...)` (and equivalent) with the subchannel being set to the 3D one.

Reply:
All but `pb_push4f` fixed due to it's type casting stuff.

Reply:
We should potentially consider to add those `push4fto` for completeness *in the future*.

Reply:
Consider it done. :+1: